lib/graphql/batch/loader.rb

```diff
@@ -126,6 +136,36 @@ def fulfilled?(key)
       promise.pending? && promise.source != self
     end
 
+    def perform_on_wait(keys)
+      # FIXME: Better name?
+      # Interface to add custom code to e.g. trigger async operations when any loader starts waiting.
```

Comment:
Could you move these comments into the method docs above the `def`? That way they show on hover, in generated docs, etc.